### PR TITLE
BAU: report coverage to Codacy

### PR DESCRIPTION
**Pull request by bjgill**
Added github authentication so we don't fall foul of the API rate limiting

### CODE REVIEW

**Comment by CharlesIC** [comment]
Looking at alphagov/verify-hub#262 - don't we need much more stuff in `build.gradle`?

File: .travis.yml [hunk 1]
```diff
@@ -1,15 +1,23 @@
...
+    - secure: "GnVoNGoi0tpNVORu/xH6iL4pGFTBtk3lxFyBSn/nrdSzNKa8VFdwcItLrNMAqDPQ0J9Mj5LhJfciRFX012uEIb+AJYYEIPMnKV+vYbjceagj9lM5X/NXvPWuS6BByE6sbV+hxGA/iSbxInIplaA2YS8HyuB1UesD12u35YXgt4sSS4i3xiLW90NZ6poY7xXiYM2ghY9dLCXqn6DggCVR+4Jm77eYvJN89Tk3bU7fnHbOMz6yOnEADUG84AtMBfEoQxuelIc2dJ6pto5A/Am6s4LKsWnSCMZsBgZ3VpIdM26a+lg68CvBIe4aHDcQc6daAVvpfcVVG3XXhNEKyHYskLFeCPxIgpXvOXTMtpPn79Dw0nSviPbznp3pQ7Jqx11ztjYLj4s5rE5AuJtD5xF8SxmVpgmQZhAMC3VdcVZ5EWkGGnno19Y2h3vU8T4btD3nqQ0eG5dHFKUPV5ifHE8u+Z0gYBVgmNPp5z5QC98bTjSubE8HnSRJBX+wpjyCyKZSjArByDzHGLKqEMMvj4GBsrAh9QcTDBPpyH6yF3HxSf1eunF8h3NAeSwVkAUrStUtthPLP1NEKc5qB7yEbBCXu5eowMsbpZ3au5hSeNVoOOJnBoOk9YDJ7i3ob9kU5c2ABBTIzNsWuVmc9O/psLUJYs6fX+u4qmSil4GDehICpxY="
 jdk:
   - oraclejdk8
   - openjdk8
   - openjdk11
+before_install:
+  - sudo apt-get install jq
+  - curl -u ida-codacy-bot:$GITHUB_PAT -LSs $(curl -u ida-codacy-bot:$GITHUB_PAT -LSs https://api.github.com/repos/codacy/codacy-coverage-reporter/releases/latest | jq -r '.assets | map({content_type, browser_download_url} | select(.content_type | contains("java-archive"))) | .[0].browser_download_url') -o codacy-coverage-reporter-assembly.jar
```
Comment (chrisclayson) [inline .travis.yml]: Do we the `-u ida-codacy-bot:$GITHUB_PAT` authentication parms? The repos queried seem to be publicly accessible.

Comment (chrisclayson) [reply .travis.yml]: Ignore me ... just re-read the PR description! 🙄

**Comment by bjgill** [comment]
The hub needs much more in its build.gradle because the hub consists of multiple projects in one repo. The MSA is just one project, so the default setup just works.

**Pull request merged**